src/components/Calculator.jsx

```diff
@@ -0,0 +1,71 @@
+/* eslint linebreak-style: 0 */
+import React, { Component } from "react";
+import "./Calculator.css";
+import { Button } from "./Button";
+import { Input } from "./Input";
+import { ClearButton } from "./ClearButton";
+import * as math from 'mathjs'
+
+class Calculator extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      input: "",
+    };
+  }
+
+  addToInput = val => {
+    this.setState({input: this.state.input + val})
+  }
+
+  handleEqual = () => {
+    this.setState({input: math.evaluate(this.state.input)})
+  }
+
+  render() {
+    return (
+      <div className="app">
+        <div className="calc-wrapper">
+          {/* eslint-disable-next-line react/destructuring-assignment */}
+          <Input input={this.state.input} />
+          <div className="row">
+            <ClearButton handleClear={() => this.setState({ input: "" })}>
+              AC
+            </ClearButton>
+            <Button handleClick={this.addToInput}>+/-</Button>
+            <Button handleClick={this.addToInput}>%</Button>
+            <Button handleClick={this.addToInput}>/</Button>
+          </div>
+          <div className="row">
+            <Button handleClick={this.addToInput}>7</Button>
+            <Button handleClick={this.addToInput}>8</Button>
+            <Button handleClick={this.addToInput}>9</Button>
+            <Button handleClick={this.addToInput}>*</Button>
+          </div>
+          <div className="row">
+            <Button handleClick={this.addToInput}>4</Button>
+            <Button handleClick={this.addToInput}>5</Button>
+            <Button handleClick={this.addToInput}>6</Button>
+            <Button handleClick={this.addToInput}>-</Button>
+          </div>
+          <div className="row">
+            <Button handleClick={this.addToInput}>1</Button>
+            <Button handleClick={this.addToInput}>2</Button>
+            <Button handleClick={this.addToInput}>3</Button>
+            <Button handleClick={this.addToInput}>+</Button>
+          </div>
+          <div className="row">
+            <div className="zero">
+              <Button handleClick={this.addToInput}>0</Button>
+            </div>
+            <Button handleClick={this.addToInput}>.</Button>
+            <Button handleClick={() => this.handleEqual()}>=</Button>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default Calculator;
```

Comment:
You did a great job by developing you're on code to make the calculator work but I  advised you to use the code already-developed, math logic needed to make work the Calculator component. as it was required in the project requirement 

- the project must have an operate.js file copied from [this one](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js). :+1: 
-  The project must have a calculate.js file copied from [this one](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js). and at the end Copy the files calculate.js and operate.js into a logic/ directory in your project. :+1: 
Analyze the files calculate.js and operate.js, be sure you understand what they do (note: you need to add [big.js](https://www.npmjs.com/package/big-js) as a dependency).
- Import the files in your Calculator component. :+1: 
- Implement the event handlers you need to use the math logic from calculate.js and operate.js in your React component. :+1: 
- Practice what you have learned until now: use state and props, pass props to child components, use lifecycle methods, and lift the state up. :+1: 
- Test your app in the browser. It should be able to perform math operations. :+1:

Reply:
@MaganaAsiati Thank you for reviewing my code. I am now going to go through the components I have improved after your suggestions.

- Added operate.js file :heavy_check_mark:
- Added calculate.js file :heavy_check_mark:
- Imported the files in Calculator component. :heavy_check_mark:
- Implemented the event handlers in React component and used the math logic from calculate.js and operate.js :heavy_check_mark:
- Used state, proprs, passed props to child components, used lifecycle methods and lifted the state up. :heavy_check_mark:
- Tested my app in the browser. Everything is working in order. :heavy_check_mark: